components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectExecutor.java

```diff
@@ -83,6 +84,41 @@ public class OpenIDConnectExecutor implements Executor {
             "nonce"};
     private static final String OIDC_DIALECT = "http://wso2.org/oidc/claim";
 
+    @Override
+    public String getName() {
+
+        return OIDC_SIGNUP_EXECUTOR;
+    }
+
+    @Override
+    public ExecutorResponse execute(FlowExecutionContext flowExecutionContext) {
+
+        try {
+            if (isInitialRequest(flowExecutionContext)) {
+                return initiateSocialSignup(flowExecutionContext);
+            }
+            return processResponse(flowExecutionContext);
+        } catch (FlowEngineException e) {
+            LOG.error("Error while executing OpenID Connect executor.", e);
+            ExecutorResponse executorResponse = new ExecutorResponse();
+            executorResponse.setResult(STATUS_ERROR);
+            executorResponse.setErrorMessage("Error while executing "+ this.getName() + ": " + e.getDescription());
```

Comment:
Some places we are throwing null for e. (Eg:-  throw handleFlowEngineServerException("State parameter mismatch.", null);) in that case the above line will cause NPE.